HW1/main.swift

```diff
@@ -7,4 +7,256 @@
 
 import Foundation
 
-print("Hello, !")
+//1. Сортировка чисел: Дан массив чисел. Напишите функцию, которая возвращает отсортированный массив в порядке возрастания без использования стандартной функции sort().
+func sort(_ array: [Int]) -> [Int] {
+    var arr = array
+    let n = array.count
+    
+    for i in 0..<n-1 {
+        for m in 0..<n-i-1 {
+            if arr[m] > arr[m+1] {
+                let temp = arr[m]
+                arr[m] = arr[m+1]
+                arr[m+1] = temp
+            }
+        }
+    }
+    return arr
+}
+
+let a = sort([55, 4, 33, 2, 11])
+print(a)
+
+//2. Проверка палиндрома: Напишите функцию, которая принимает строку и возвращает true, если строка является палиндромом, и false в противном случае.
+func isPalindrome(string:String) -> Bool {
+    let reverseString = String(string.reversed())
+    
+    if string != "" && string == reverseString {
+        print("Это слово полиндром")
+        return true
+    } else {
+        print("Это не полиндром")
+        return false
+    }
+}
+
+//isPalindrome(string: "mam")
+
+//3. 3. Сумма элементов массива: Дан массив целых чисел. Напишите функцию, которая возвращает сумму всех элементов массива.
+func sumOf(numbers: [Int]) -> Int {
+    var sumNumber = 0
+    
+    for i in 1...numbers.count {
+        sumNumber += numbers[i-1]
+    }
+    return sumNumber
+}
+
+let check = sumOf(numbers: [1,33,554])
+print(check)
+
+//4. Конвертация температуры: fahrenheit = (celsius * 9.0/5.0) + 32.0. Напишите функцию, которая принимает значение температуры в градусах Цельсия и возвращает его эквивалент в градусах Фаренгейта.
+func fromCelToFahr(num: Double) -> Double {
+    let fahrenheit = num * 9/5 + 32
+    return fahrenheit
+}
+
+let fahr = fromCelToFahr(num: 32)
+print(fahr)
+
+// 5. Записная книжка: Создайте массив, который содержит имя, телефон и электронную почту. Затем напишите функцию для поиска записи по имени.
+//let notes = ["David", "ber.dev@yandex.ru", "+7932"]
+//let name = readLine()
+//
+//func search(notes: [String]) {
+//
+//    for note in notes {
+//        if note == name {
+//            print("Имя \(name ?? "") найдено")
+//            break
+//        } else {
+//            print("Имя \(name ?? "") не найдено")
+//        }
+//    }
+//}
+
+//search(notes: notes)
+
+
+//6. Перестановка элементов массива: Напишите функцию, которая принимает массив и возвращает новый массив, где первый элемент становится последним, а последний — первым. Остальные элементы остаются на своих местах.
+
+func changeArrays(array: [Int]) -> [Int] {
+    var arr = array
+    let count = array.count
+    
+    let c = arr[0]
+    arr[0] = arr[count-1]
+    arr[count-1] = c
+    
+    print(arr)
+    return arr
+}
+
+changeArrays(array: [3,2,22])
+
+//7. Форматирование строки: Напишите функцию, которая принимает строку и делает первую букву каждого слова заглавной.
```

Comment:
func upperFirst(_ string: String) -> String {
    let words = string.split(separator: " ")
    guard words.count > 1 else { return "We have one word!" }
    var stringFormat = String()

    for element in words {
        let firstChar = element.prefix(1).uppercased()
        let restoreWords = element.dropFirst()
        stringFormat += firstChar + restoreWords + " "
    }


    return stringFormat
}